From fca8aecb20fc76463f06a3def040b6e39303bfa1 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:56 -0400
Subject: [ethernet] mlx4: Fix sparse warning

Message-id: <df57a813ff7566129ba470f81df5264edf635fa2.1410425016.git.avadai@redhat.com>
Patchwork-id: 93694
O-Subject: [Patch RHEL7.1 V1 060/197] net/mlx4_core: Fix sparse warning
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Eyal Perry <eyalpe@mellanox.com>

This patch force conversion to u32 to fix the following sparse warning:
drivers/net/ethernet/mellanox/mlx4/fw.c:1822:53: warning: restricted __be32
degrades to integer

Casting to u32 is safe here, because token will be returned as is
from the hardware without any modification.

Signed-off-by: Eyal Perry <eyalpe@mellanox.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 28d222bbaa5122fb4bb0e607e39ab149a010e587)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/fw.c b/drivers/net/ethernet/mellanox/mlx4/fw.c
index 93bee66..d2af281 100644
--- a/drivers/net/ethernet/mellanox/mlx4/fw.c
+++ b/drivers/net/ethernet/mellanox/mlx4/fw.c
@@ -1890,7 +1890,8 @@ void mlx4_opreq_action(struct work_struct *work)
    err = EINVAL;
    break;
   }
-  err = mlx4_cmd(dev, 0, ((u32) err | cpu_to_be32(token) << 16),
+  err = mlx4_cmd(dev, 0, ((u32) err |
+     (__force u32)cpu_to_be32(token) << 16),
           1, MLX4_CMD_GET_OP_REQ, MLX4_CMD_TIME_CLASS_A,
           MLX4_CMD_NATIVE);
   if (err) {
-- 
1.7.1